src/3p-frame.js

```diff
@@ -109,8 +109,33 @@ export function getIframe(parentWindow, element, opt_type) {
 export function listen(iframe, typeOfMessage, callback) {
   var win = iframe.ownerDocument.defaultView;
   var origin = parseUrl(getBootstrapBaseUrl(win)).origin;
+  listen_(iframe, origin, typeOfMessage, callback);
+}
+
+/**
+ * Allows listening for message from the iframe.
+ * @param {!Element} iframe
+ * @param {string} typeOfMessage
+ * @param {function(!Object)} callback Called when a message of this type
+ *     arrives for this iframe.
+ */
+export function listenNonSensitive(iframe, typeOfMessage, callback) {
+  listen_(iframe, '*', typeOfMessage, callback);
+}
+
+/**
+ * Allows listening for message from the iframe, optionally with filtering on
+ * origin. Origin can be specified as '*' to avoid origin check.
+ * @param {!Element} iframe
+ * @param {string} origin
+ * @param {string} typeOfMessage
+ * @param {function(!Object)} callback Called when a message of this type
+ *     arrives for this iframe.
+ */
+export function listen_(iframe, origin, typeOfMessage, callback) {
+  let win = iframe.ownerDocument.defaultView;
   win.addEventListener('message', function(event) {
-    if (event.origin != origin) {
+    if (origin != '*' && event.origin != origin) {
```

Comment:
"OK" as in let's keep '*'? Or let's get origin from URL assuming that you will always convert srcdoc to a data URL? I likewise judged '*' this to be ok since we always check on source window.